File: python/credit_check.py
from posixpath import split
import functools

def credit_check(string):
    split_str = list(map(int,str(string)))

    for i in range(0, len(split_str), 2):
        if split_str[i] * 2 > 9:
            split_str[i] = functools.reduce(lambda agg, item : agg + item, list(map(int,str(split_str[i] * 2))))
        else :
            split_str[i] = split_str[i] * 2
    return "The number is valid!" if functools.reduce(lambda agg, item : agg + item, split_str) % 10 == 0 else "The number is invalid!"

File: js/creditCheck.js
exports.creditCheck = function (str) {
    let strSplit = str.split('')
    let strToNumArr = strSplit.map(elem => parseInt(elem))

    for (let i = 0; i < strToNumArr.length; i = i + 2) {
        if (strToNumArr[i] * 2 > 9) {
            strToNumArr[i] = (strToNumArr[i] * 2).toString().split('').map(elem => parseInt(elem)).reduce((a, b) => a + b)
        } else {
            strToNumArr[i] = strToNumArr[i] * 2
        }
    }

    return strToNumArr.reduce((a, b) => a + b) % 10 === 0 ? "The number is valid!" : "The number is invalid!"
}


